src/LowercaseSentenceTokenizer.java

```diff
@@ -29,8 +30,28 @@ public class LowercaseSentenceTokenizer implements Tokenizer {
    * @return a list of tokens, where each token is a word or a period
    */
   public List<String> tokenize(Scanner scanner) {
-    // TODO: Implement this function to convert the scanner's input to a list of words and periods
-    return null;
+    List<String> tokens = new ArrayList<>();
+    
+    // Reading all text through this scanner
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine().toLowerCase();
+      
+      // Split the line by spaces
+      String[] words = line.split("\\s+");
```

Comment:
This works well! However, `Scanner` by default splits on whitespace. Can you think of how you could use `scanner.next` instead of `scanner.nextLine` to simplify your code?